packages/server-wallet/src/models/objective.ts

```diff
@@ -191,9 +191,12 @@ export class ObjectiveModel extends Model {
         .returning('*')
         .first();
 
-      const model = await query
-        .onConflict('objectiveId')
-        .merge({status: preApproved ? 'approved' : 'pending'});
+      // Knex does not like properties with undefined values so we strip them out
+      const override = _.pickBy({status: approvalStatus, waitingFor}, _.identity);
+      // This allows for for the insert method to override the existing status or waitingFor if desired
+      const model = _.isEmpty(override)
```

Comment:
Curious, is the conditional necessary here? Does `const model = await query.onConflict('objectiveId').merge(override);` work with a empty `override` object?

Reply:
Yup, unfortunately `knex` fails when passed in an empty objective.
 ``` 
Empty .update() call detected! Update data does not contain any values to update. This will result in a faulty query. Table: objectives. 
```